client/cmdhfmf.c

```diff
@@ -1218,9 +1222,11 @@ int CmdHF14AMfChk(const char *Cmd)
 	}
 	printf("\n");
 
-	bool foundAKey = false;
-	uint32_t max_keys = keycnt > USB_CMD_DATA_SIZE / 6 ? USB_CMD_DATA_SIZE / 6 : keycnt;
-	if (SectorsCnt) {
+	bool     foundAKey = false;
+	uint32_t max_keys  = keycnt > USB_CMD_DATA_SIZE / 6 ? USB_CMD_DATA_SIZE / 6 : keycnt;
+
+    // !SingleKey, so all key check
+	if ((SectorsCnt) && !SingleKey) { 

```

Comment:
Hmmm, is there a possibility that `SectorsCnt == 0` ?

Reply:
I will have a closer look.  My initial thought was SectorsCnt = based on card type (so maybe), or the single block check, so set to >= 1.  The && !SingleKey was added due to the fact that I set the SectorCnt to a value (so I could reuse the code to write to the emu.

Reply:
On closer look at this, if * is provided then SectorsCnt should be valid for the card size.
On testing SectorsCnt = ParamCardSizeSectors(param_getchar(Cmd + 1, 0)); will default to a 16 Sectors if an invalid value e.g. *55
  
If * was not provided then the block needs to be provided for SingleKey to be true, so added a check on the else to catch if something was missing/went wrong.